spec/controllers/portal/portal_controller_spec.rb

```diff
@@ -1,6 +1,28 @@
 require "rails_helper"
 
 RSpec.describe Portal::PortalController, type: :controller do
+  describe "#current_intake" do
+    let(:session_intake) { create :intake }
+    let(:client) { create :client, intake: (create :intake) }
+
+    before do
+      session[:intake_id] = session_intake.id
+    end
+
+    context "when the client is authenticated" do
+      before do
+        sign_in client, scope: :client
+      end
+      it "sets the current intake to that clients intake" do
+        expect(subject.current_intake).to eq client.intake
+      end
+    end
+    context "when the client is not authenticated" do
+      it "uses the intake_id from the session to determine the current_intake (like in unauthenticated intake flow" do
+        expect(subject.current_intake).to eq session_intake
```

Comment:
It'd be fine with me if `current_intake` were defined as nil in this controller until you're logged in, since it's not really relevant. An option for aggressive cleanup. Shrug; not a blocker.

Reply:
Updated this.